### DISCUSSION TITLE
Change hovernet losses to MaskedLoss?

### DISCUSSION THREAD

**Discussion by OmarAshkar** [Project-MONAI/MONAI#6906] [Q&A]
Hi,

I have a challenge in my data of a lot of missing labels. I am wondering if I can use the concept of MaskedLoss here? What I understand that 0's will not be accounted for in the loss.

I was thinking to  change these 2 lines of DiceLoss and CrossentopyLoss
https://github.com/Project-MONAI/MONAI/blob/2daabf917a1e6afbc738d22db602f4859672a90d/monai/apps/pathology/losses/hovernet_loss.py#L62

I need some guidance on how to change those lines please.

Thanks alot!

**Comment by JohnMasoner** [comment]
hey @OmarAshkar ,

MaskedLoss may not be reasonable as you describe it.  The MaskedLoss def:

> region with mask 1 will keep the original value, region with 0 mask will be converted to 0.

but your data of a lot of missing labels.

For the problems with your data, my suggestion you can either set the label to 0, or remove the data with missing labels, or use a semi-supervised training.

Mason

**Reply by JohnMasoner** [reply]
I'm not sure that methods is a better fit for your data. But my suggestion is that you can use labeled data for training.  bc the smei-supervised need a base model has a base dice score.

**Reply by OmarAshkar** [reply]
Thanks @JohnMasoner for your quick and comprehensive response. I didn't get exactly the idea of weakly supervised. Is there is a monai tutorials and implementation of it?

**Reply by JohnMasoner** [reply]
As far as I can tell monai does not currently implement this feature. Maybe you can try [Unimatch(CVPR2023)](https://github.com/LiheYoung/UniMatch/tree/main/more-scenarios/medical). (I recommend this repo because I have implemented their work, the repo had a good performance.)
